tests/test_harvest_bookmarks.py

```diff
@@ -632,9 +575,67 @@ def do_test(self, conn_id):
                         if matching:
                             break
                     self.assertTrue(matching)
-
-                # verify all data from 2nd sync >= 1st bookmark
+                
                 if stream not in no_replication_key:
+
+                    stream_bookmark_keys = self.expected_replication_keys().get(stream, set())
+                    assert len(stream_bookmark_keys) == 1  # There shouldn't be a compound replication key
+
+                    stream_bookmark_key = next(iter(stream_bookmark_keys))
+                    target_value = first_max_bookmarks.get(
+                        stream, {None: None}).get(stream_bookmark_key)
+                    target_min_value = first_min_bookmarks.get(
+                        stream, {None: None}).get(stream_bookmark_key)
+                    final_bookmark = second_max_bookmarks.get(
+                        stream, {None: None}).get(stream_bookmark_key)
+
+                    try:
+                        # Attempt to parse the bookmark as a date
+                        if state_value:
+                            if isinstance(state_value, str):
+                                state_value = self.local_to_utc(parse(state_value))
+                            if isinstance(state_value, int):
+                                state_value = self.local_to_utc(dt.utcfromtimestamp(state_value))
```

Comment:
This code is duplicated, create a function for this.

Reply:
Created common function to parse datetime values.